File: Client/src/app/models/CookedRecipeCalledIngredientDTO.ts
import { CalledIngredientDTO } from "./CalledIngredientDTO";
import { KitchenProductDTO } from "./KitchenProductDTO";
import { UnitTypeDTO } from "./UnitTypeDTO";

export class CookedRecipeCalledIngredientDTO {
  id?: number = undefined;
  cookedRecipeId?: number = undefined;
  calledIngredient?: CalledIngredientDTO = undefined;
  kitchenProduct?: KitchenProductDTO = undefined;
  kitchenProductId?: number = undefined;
  name?: string = undefined;
  kitchenUnitType?: UnitTypeDTO = undefined;
  amount?: number = undefined;
  kitchenProductSearchItems?: KitchenProductDTO[] = undefined;
}

File: API/ContainerNinja.Contracts/DTO/CookedRecipeCalledIngredientDTO.cs
using ContainerNinja.Contracts.Data.Entities;
using ContainerNinja.Contracts.Enum;

namespace ContainerNinja.Contracts.DTO
{
    public class CookedRecipeCalledIngredientDTO
    {
        public int Id { get; set; }
        public int CookedRecipeId { get; set; }
        public CalledIngredientDTO? CalledIngredient { get; set; }
        public KitchenProductDTO? KitchenProduct { get; set; }
        public int? KitchenProductId { get; set; }
        public string Name { get; set; }
        public KitchenUnitType KitchenUnitType { get; set; }
        public float? Amount { get; set; }
    }
}